frame/referenda/src/types.rs

```diff
@@ -512,6 +514,18 @@ impl Debug for Curve {
 	}
 }
 
+/// General information about the item referring to the metadata.
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct Metadata<Schema: Eq + PartialEq + Debug + Encode + Decode + TypeInfo + Clone> {
+	/// The schema/s descriptor of the data the preimage `hash` referring to.
+	/// e.g. enum of `IpfsJsonV1` (the hash of an off-chain IPFS json file),
+	/// `BinJsonV2` (on-chain json dump).
+	pub(super) schema: Schema,
```

Comment:
as we talked about in DM, it does not make sense to me to leak this into the pallet. The pallet should just have a hash.

I feel like whatever schema thing you are looking to build here should really be a layer inside the preimage pallet.

Reply:
Makes sense to me now.

By meaning the schema should be a property of the uploaded preimage. If referendum needs its own list of possible schemas, they can be whitelisted or have its own preimage instance. And it prevents the duplication of the schema type everywhere. 

I will keep this PR free from the schema type idea.